client/components/Editor/index.jsx

```diff
@@ -144,29 +185,56 @@ class Editor extends Component {
             case 'slides':
                 return (
                     <Slides
-                        scenarioId={this.props.match.params.id}
+                        scenarioId={scenarioId || this.props.match.params.id}
                         updateEditorMessage={this.updateEditorMessage}
                     />
                 );
             case 'preview':
-                return <Scenario scenarioId={this.props.match.params.id} />;
+                return (
+                    <Scenario
+                        scenarioId={scenarioId || this.props.match.params.id}
+                    />
+                );
             default:
                 return null;
         }
     }
 
+    getAllTabs(scenarioId) {
+        if (!scenarioId) return {};
+        let copyId;
+        switch (scenarioId) {
+            case 'new':
+                return {
+                    moment: this.getTab('moment', 'new')
+                };
+            case 'copy':
+                copyId = String(this.props.location.state.scenarioCopy.id);
+                return {
+                    moment: this.getTab('moment', copyId),
+                    slides: this.getTab('slides', copyId),
+                    preview: this.getTab('preview', copyId)
+                };
+            default:
+                return {
+                    moment: this.getTab('moment'),
+                    slides: this.getTab('slides'),
+                    preview: this.getTab('preview')
+                };
+        }
+    }
+
     getSubmitCallback() {
         let endpoint, method;
         const scenarioId = this.props.match.params.id;
 
-        if (scenarioId === 'new') {
+        if (this.props.isNewScenario || scenarioId === 'copy') {
```

Comment:
We had this `isNewScenario` prop already thanks to our component in routes, I'm just taking advantage of it here.